### PR TITLE
feat(biome_css_analyzer): implement selector-pseudo-class-no-unknown

### LINKED ISSUE #2625: 📎  Implement selector-pseudo-class-no-unknown
## Description

Implement [selector-pseudo-class-no-unknown](https://stylelint.io/user-guide/rules/selector-pseudo-class-no-unknown)

> [!IMPORTANT]
> - Please skip implementing options for now since we will evaluate users actually want them later.
> - Please ignore handling extended CSS language cases such as `sass` and `less`.
> - Please skip custom function since we haven't syntax model yet.

**Want to contribute?** Lets you know you are interested! We will assign you to the issue to prevent several people to work on the same issue. Don't worry, we can unassign you later if you are no longer interested in the issue! Read our [contributing guide](https://github.com/biomejs/biome/blob/main/CONTRIBUTING.md) and [analyzer contributing guide](https://github.com/biomejs/biome/blob/main/crates/biome_analyze/CONTRIBUTING.md).

### PR DESCRIPTION
**Pull request by tunamaguro**
## Summary

close #2625
implement [selector-pseudo-class-no-unknown](https://stylelint.io/user-guide/rules/selector-pseudo-class-no-unknown/)

## Test Plan

Add test cases from `stylelint` : https://github.com/stylelint/stylelint/blob/main/lib/rules/selector-pseudo-class-no-unknown/__tests__/index.mjs

### CODE REVIEW

File: crates/biome_css_analyze/src/lint/nursery/no_unknown_pseudo_class_selector.rs [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+    while let Some(prev) = &prev_element {
+        let maybe_selector: Option<CssPseudoElementSelector> = prev.clone().cast();
+        if let Some(selector) = maybe_selector.as_ref() {
+            return WEBKIT_SCROLLBAR_PSEUDO_ELEMENTS.contains(&selector.text().trim_matches(':'));
+        };
+        prev_element = prev.prev_sibling();
+    }
+
+    false
+}
```
Comment (tunamaguro) [inline crates/biome_css_analyze/src/lint/nursery/no_unknown_pseudo_class_selector.rs:83]: To determine whether a `::-webkit-scrollbar CSS pseudo-element` has a selector, I'm retrieving each previous element one by one, but I feel there might be a performance issue.

File: crates/biome_css_analyze/src/lint/nursery/no_unknown_pseudo_class_selector.rs [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+        recommended: false,
+        sources: &[RuleSource::Stylelint("selector-pseudo-class-no-unknown")],
+    }
+}
+declare_node_union! {
+    pub AnyPseudoLike = CssPseudoClassFunctionCompoundSelector|CssPseudoClassFunctionCompoundSelectorList|CssPseudoClassFunctionIdentifier
+                        |CssPseudoClassFunctionNth|CssPseudoClassFunctionRelativeSelectorList|CssPseudoClassFunctionSelector
+                        |CssPseudoClassFunctionSelectorList|CssPseudoClassFunctionValueList|CssPseudoClassIdentifier
+                        |CssBogusPseudoClass|CssPageSelectorPseudo
+}
```
Comment (togami2864) [inline crates/biome_css_analyze/src/lint/nursery/no_unknown_pseudo_class_selector.rs:70]:
```suggestion
declare_node_union! {
  pub AnyPseudoLike =
      CssPseudoClassFunctionCompoundSelector
      | CssPseudoClassFunctionCompoundSelectorList
      | CssPseudoClassFunctionIdentifier
      | CssPseudoClassFunctionNth
      | CssPseudoClassFunctionRelativeSelectorList
      | CssPseudoClassFunctionSelector
      | CssPseudoClassFunctionSelectorList
      | CssPseudoClassFunctionValueList
      | CssPseudoClassIdentifier
      | CssBogusPseudoClass
      | CssPageSelectorPseudo
}
```

File: crates/biome_css_analyze/src/lint/nursery/no_unknown_pseudo_class_selector.rs [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+
+fn is_webkit_pseudo_class(node: &AnyPseudoLike) -> bool {
+    let mut prev_element = node.syntax().parent().and_then(|p| p.prev_sibling());
+    while let Some(prev) = &prev_element {
+        let maybe_selector: Option<CssPseudoElementSelector> = prev.clone().cast();
```
Comment (togami2864) [inline crates/biome_css_analyze/src/lint/nursery/no_unknown_pseudo_class_selector.rs:75]:
```suggestion
        let maybe_selector = CssPseudoElementSelector::cast_ref(prev);
```

File: crates/biome_css_analyze/tests/specs/nursery/noUnknownPseudoClassSelector/invalid.css [hunk 4]
```diff
@@ -0,0 +1,15 @@
+a:unknown { }
+a:Unknown { }
+a:uNkNoWn { }
+a:UNKNOWN { }
+a:pseudo-class { }
+body:not(div):noot(span) {}
+a:unknown::before { }
+a,\nb > .foo:error { }
```
Comment (togami2864) [inline crates/biome_css_analyze/tests/specs/nursery/noUnknownPseudoClassSelector/invalid.css:8]:
```suggestion
a,
b > .foo:error { }
```

File: crates/biome_css_analyze/tests/specs/nursery/noUnknownPseudoClassSelector/valid.css [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+:matches(section, article, aside, nav) h1 { }
+a:has(> img) { }
+section:has(h1, h2, h3, h4, h5, h6) { }
+:root { }
+p:has(img):not(:has(:not(img))) { }
+div.sidebar:has(*:nth-child(5)):not(:has(*:nth-child(6))) { }
+div :nth-child(2 of .widget) { }
+a:hover::before { }
+a:-moz-placeholder { }
+a,\nb > .foo:hover { }
```
Comment (togami2864) [inline crates/biome_css_analyze/tests/specs/nursery/noUnknownPseudoClassSelector/valid.css:19]:
```suggestion
a,
b > .foo:hover { }
```

File: crates/biome_css_analyze/src/lint/nursery/no_unknown_pseudo_class_selector.rs [hunk 6]
```diff
@@ -0,0 +1,218 @@
...
+    ///
+    /// ```css
+    /// input:-moz-placeholder {}
+    /// ```
+    ///
+    pub NoUnknownPseudoClassSelector {
+        version: "next",
+        name: "noUnknownPseudoClassSelector",
+        language: "css",
+        recommended: false,
```
Comment (togami2864) [inline crates/biome_css_analyze/src/lint/nursery/no_unknown_pseudo_class_selector.rs:61]:
```suggestion
        recommended: true,
```

**Review by togami2864 — COMMENTED** [review]
Thank you! I left some suggestions.

File: crates/biome_css_analyze/src/lint/nursery/no_unknown_pseudo_class_selector.rs (hunk 3 above)
Comment (tunamaguro) [reply crates/biome_css_analyze/src/lint/nursery/no_unknown_pseudo_class_selector.rs]: Thank you for the suggestions. I fixed it in https://github.com/biomejs/biome/pull/3034/commits/4eae3f005a07fbe712fc7d215a8daf70e8dcc8b6

File: crates/biome_css_analyze/tests/specs/nursery/noUnknownPseudoClassSelector/invalid.css (hunk 4 above)
Comment (tunamaguro) [reply crates/biome_css_analyze/tests/specs/nursery/noUnknownPseudoClassSelector/invalid.css]: I fixed it in https://github.com/biomejs/biome/pull/3034/commits/31441057cb0fca66508afb155054ca170fe32894 and update snapshots

File: crates/biome_css_analyze/tests/specs/nursery/noUnknownPseudoClassSelector/valid.css (hunk 5 above)
Comment (tunamaguro) [reply crates/biome_css_analyze/tests/specs/nursery/noUnknownPseudoClassSelector/valid.css]: I fixed it in https://github.com/biomejs/biome/commit/31441057cb0fca66508afb155054ca170fe32894 and update snapshots

File: crates/biome_css_analyze/src/lint/nursery/no_unknown_pseudo_class_selector.rs (hunk 6 above)
Comment (tunamaguro) [reply crates/biome_css_analyze/src/lint/nursery/no_unknown_pseudo_class_selector.rs]: I fixed it in https://github.com/biomejs/biome/pull/3034/commits/83e8e9f29c8234e5da8fdb75a4744984c49054b2

File: crates/biome_css_analyze/src/lint/nursery/no_unknown_pseudo_class_selector.rs [hunk 7]
```diff
@@ -0,0 +1,226 @@
...
+    /// ```css
+    /// :not(p) {}
+    /// ```
+    ///
+    /// ```css
+    /// input:-moz-placeholder {}
+    /// ```
+    ///
+    pub NoUnknownPseudoClassSelector {
+        version: "1.8.0",
```
Comment (togami2864) [inline crates/biome_css_analyze/src/lint/nursery/no_unknown_pseudo_class_selector.rs:58]:
```suggestion
        version: "next",
```

**Review by togami2864 — APPROVED** [review]
I left one small nit but LGTM

File: crates/biome_css_analyze/src/lint/nursery/no_unknown_pseudo_class_selector.rs (hunk 7 above)
Comment (tunamaguro) [reply crates/biome_css_analyze/src/lint/nursery/no_unknown_pseudo_class_selector.rs:58]: I was not aware of this. Thanks for the suggestion!

**Pull request merged**